pypesto/optimize/optimizer.py

```diff
@@ -106,6 +106,7 @@ def wrapped_minimize(
             id=id,
             x_names=[problem.x_names[ix] for ix in problem.x_free_indices],
             options=history_options,
+            amici_objective=isinstance(objective, AmiciObjective)
```

Comment:
I'd strongly prefer to keep AMICI out of pypesto.{optimize,history}.
How about something like:
```python
from pypesto.history.generate import create_history
class ObjectiveBase:
   def create_history(...):
       create_history(...)
```
and doing the amici-specific magic in `AmiciObjective.create_history`?

Reply:
Maybe break create_history in two parts: without storage file and with storage file, and implement both in ObjectiveBase (then pypesto.history.generate isn't needed)? Then these can be re-implemented in Objectives as needed.

Reply:
> then pypesto.history.generate isn't needed

That could/should be done in the medium term. For now, I'd just leave it there and re-use in ObjectiveBase.create_history, just in case somebody is actually using pypesto.history.generate.create_history directly.